File: motion_planning/scripts/TrajectoryPublisher.py
#!/usr/bin/env python

import rospy
from nav_msgs.msg import Path

class TrajectoryPublisher:
    """
    Publishes trajectory messages describing the robot's desired trajectory.

    The robot's trajectory, composed of an array of positions, orientations, and times, is represented by nav_msgs/Path messages published to the motion_planning/trajectory topic.

    Attributes:
        trajectory_pub: Publisher to motion_planning/trajectory. Publishes nav_msgs/Path messages.
    """

    def __init__(self):
        """Initialize TrajectoryPublisher instance."""
        self.trajectory_pub = rospy.Publisher('motion_planning/trajectory', Path, queue_size=10)

    def publishTrajectory(self, trajectory):
        """
        Publish a nav_msgs/Path message.

        Uses the trajectory_pub Publisher to publish to motion_planning/trajectory.

        Args:
            trajectory: The nav_msgs/Path message.
        """
        self.trajectory_pub.publish(trajectory)

File: motion_planning/src/TrajectoryPublisher.cc
#include "motion_planning/TrajectoryPublisher.hh"

namespace MuddSub::MotionPlanning
{
  TrajectoryPublisher::TrajectoryPublisher(ros::NodeHandle n):
    n_(n)
  {
    trajectory_pub = n_.advertise<nav_msgs::Path>("motion_planning/trajectory", 1000);
  }

  void TrajectoryPublisher::publishTrajectory(nav_msgs::Path& trajectory)
  {
    trajectory_pub.publish(trajectory);
  }
}
